static/js/ajaxUpdate.js

```diff
@@ -1,51 +1,46 @@
-export async function ajaxDelete(options) {
+import getCsrfToken from './getCsrfToken.js';
+
+export function ajaxDelete(options) {
     const fetchInit = {
         method: 'DELETE',
+        headers: {
+            'X-CSRFToken': getCsrfToken(),
+            'Accept': options.headersAccept,
+            'Content-Type': options.headersContentType,
+        },
     };
-    return ajax(options.submit_url, fetchInit);    
+    return ajax(options.submit_url, fetchInit);
 }
-export async function ajaxUpdate(data, options) {
+
+export function ajaxUpdate(data, options) {
     const fetchInit = {
         method: 'POST',
+        headers: {
+            'X-CSRFToken': getCsrfToken(),
+            'Accept': options.headersAccept,
+            'Content-Type': options.headersContentType,
+        },
         body: BODY_ENCODE_FUNCTIONS[options.bodyEncode](data),
     };
-    return ajax(options.submit_url, fetchInit);    
+    return ajax(options.submit_url, fetchInit);
 }
 
 async function ajax(url, fetchInit) {
-    let {default:getCsrfToken} = await import('./getCsrfToken.js');
-    // make our request to udpate the value
-    return new Promise((resolve, reject) => {
-        
-        fetch(
-            url,
-            Object.assign({
-                headers: {
-                    'X-CSRFToken': getCsrfToken(),
-                }
-            }, fetchInit)
-        ).then(function(response) {
-            // function that will run on promise resolve;
-            if(response.ok) {
-                resolve(response.status, response);
-            } else {
-                reject(response.statusText, response.status, response)
-            }
-        }).catch(function(err) {
-            reject(err.message, err)
-        });
-    });
+    const response = await fetch(url, fetchInit)
+        .then(res => res.json()) // pull out JSON from response returned
+        .then(json => console.log(json)) // json only returns {"status": "ok"} if all is well
```

Comment:
I'd rather we did not console log by default

Reply:
Not the final handling desired, yes.